misaligned.cpp

```diff
@@ -1,21 +1,52 @@
-#include <iostream>
-#include <assert.h>
+#include "misaligned.h"
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
-    for(i = 0; i < 5; i++) {
-        for(j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+namespace misaligned
+{
+
+    const char *majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+    int numberOfMajorColors = sizeof(majorColor) / sizeof(majorColor[0]);
+    const char *minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+    int numberOfMinorColors = sizeof(minorColor) / sizeof(minorColor[0]);
+
+    int Misaligned::printColorMap()
+    {
+        int i = 0, j = 0;
+        for (i = 0; i < 5; i++)
+        {
+            for (j = 0; j < 5; j++)
+            {
+                std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+            }
         }
+        return i * j;
     }
-    return i * j;
+
+    int Misaligned::GetPairNumberFromColor(MajorColor major, MinorColor minor)
+    {
+        return major * 5 + minor;
+    }
+
+    int Misaligned::expectNumberFromColor(MajorColor major, MinorColor minor)
+    {
+        return major * numberOfMinorColors + minor + 1;
+    }
+}
+
+void testPairToNumber(misaligned::MajorColor major, misaligned::MinorColor minor)
+{
+    misaligned::Misaligned *l_object{};
+    int pairNumber = l_object->GetPairNumberFromColor(major, minor);
+    std::cout << "Got pair number " << pairNumber << std::endl;
+    assert(pairNumber == l_object->expectNumberFromColor(major, minor));
 }
 
-int main() {
-    int result = printColorMap();
+int main()
+{
+    misaligned::Misaligned l_object;
+    int result = l_object.printColorMap();
     assert(result == 25);
+    testPairToNumber(misaligned::BLACK, misaligned::ORANGE);

```

Comment:
These test cases do not effectively test the color map. Please add test cases for pair numbers where major color ends and another major color starts such as 1, 5,6, 10,11 etc.